index.js

```diff
@@ -69,6 +69,18 @@
     };
   };
 
+  //  functionName :: f -> String
+  var functionName = function(f) {
+    // String(x => x) evaluates to "x => x", so the pattern may not match.
+    var match = String(f).match(/^function (\w+)/);
+    return match == null ? '' : match[1];
+  };
+
+  //  constructorName :: x -> String
```

Comment:
I think the type is best represented as `{ constructor :: Function } -> String`.